src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,190 @@
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "delete":
+            String deletedTaskString = taskList.getTask(Integer.parseInt(args[1]) - 1).toString();
+            taskList.removeTask(Integer.parseInt(args[1]) - 1);
+            ui.print("Deleted Task:", " " + deletedTaskString,
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "find":
+            ui.print(taskList.findTasksContaining(args[1]), "Here are the matching tasks in your list:");
+            return 1;
+        default:
+            System.out.println("Unknown Command");
+            return 0;
+        }
+    }
+
+    /**
+     * Validates a given String and argument list based on the given action and task list. <br>
+     * A DukeException is thrown when the String or argument do not fit the corresponding <br>
+     * action and task list provided. <br>
+     * <br>
+     *
+     * @param inputString The string to be checked.
+     * @param action The type action to be checked with.
+     * @param args The argument list to be checked.
+     * @param taskList The list of task to be checked with.
+     */
+
+    public void validate(String inputString, String action, String[] args, TaskList taskList) throws DukeException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/M/yyyy");
+        switch(action) {
+        case "list":
+            if (args.length > 1) {
+                if (args[1].equals("/on") || args[1].equals("/before") || args[1].equals("/after")) {
+                    try {
+                        LocalDate.parse(args[2], formatter);
+                    } catch (DateTimeParseException | ArrayIndexOutOfBoundsException e) {
+                        throw new DukeException("Provide the date in the format dd-mm-yyyy!");
+                    }
+                }
+            }
+            break;
+        case "mark":
+        case "unmark":
+        case "delete":
+            if (args.length == 1) {
+                throw new DukeException("Task ID must be provided!");
+            }
+            try {
+                Integer.parseInt(args[1]);
+            } catch (NumberFormatException e) {
+                throw new DukeException("Task ID must be an integer!");
+            }
+            if (taskList.getSize() == 0) {
+                throw new DukeException("Task List is empty!");
+            }
+            if (Integer.parseInt(args[1]) > taskList.getSize()) {
+                throw new DukeException("Task ID out of range!");
+            }
+            break;
+        case "todo":
+            if (args.length == 1) {
+                throw new DukeException("Task Name must be provided!");
+            }
+            break;
+        case "deadline":
+        case "event":
+            String flag = action.equals("deadline") ? "/by" : "/at";
```

Comment:
I feel that `flag` isnt the right name here, consider renaming `flag` to be more descriptive.
```suggestion
            String delimiter = action.equals("deadline") ? "/by" : "/at";
```